app/models/experiment_template.py

```diff
@@ -88,5 +98,7 @@ async def validate_datasets(self, dataset_ids: list[int]) -> bool:
 
         return all(checks)
 
-    def validate_env_vars(self, env_vars: dict[str, str]) -> bool:
-        return set(self.envs_required).issubset(env_vars)
+    def validate_env_vars(self, env_vars: list[EnvironmentVarValue]) -> bool:
+        exp_env_vars = set([env.name for env in env_vars])
```

Comment:
Expand the variable names to full text. It is not clear what does `exp_env_vars` and `req_env_vars` mean.